multimodal/src/autogluon/multimodal/matcher.py

```diff
@@ -1461,7 +1590,21 @@ def extract_embedding(
         by the neural network's architecture.
         """
         if signature is None:
-            signature = QUERY
+            if self._query or self._response:
+                if isinstance(data, list):
```

Comment:
I see there are several assertion statements to check if data is a list. My question is, other than list/dict/df, will other data types be expected? Is it possible that users just give a None, or a str, or a float?

Reply:
In matching, we currently don't assume other data types. A None, str, or float would encounter error when being converted to dataframe.

Reply:
@zhiqiangdon we should really use DataFrame format. Let's revise it in the next PR. It's more urgent to get the API aligned for now.

Reply:
@sxjscience The `extract_embedding` in matcher should support passing a list, dict, or a dataframe. This is aligned with the predictor.